packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/InAppPurchasePluginTests.m

```diff
@@ -332,41 +354,39 @@ - (void)testAddPaymentSuccessWithPaymentDiscount {
     }
   };
 
-  FIAPaymentQueueHandler *mockHandler = OCMClassMock(FIAPaymentQueueHandler.class);
-  OCMStub([mockHandler addPayment:[OCMArg any]]).andReturn(YES);
-  self.plugin.paymentQueueHandler = mockHandler;
+  TestPaymentQueueHandler *handler = [TestPaymentQueueHandler alloc];
+  self.plugin.paymentQueueHandler = handler;
+
+  __block NSInteger addPaymentTimes = 0;
+  handler.addPaymentStub = ^(SKPayment *payment) {
+    if (@available(iOS 12.2, *)) {
+      SKPaymentDiscount *discount = payment.paymentDiscount;
+      XCTAssertEqual(discount.identifier, @"test_identifier");
+      XCTAssertEqual(discount.keyIdentifier, @"test_key_identifier");
+      XCTAssertEqualObjects(
+          discount.nonce,
+          [[NSUUID alloc] initWithUUIDString:@"4a11a9cc-3bc3-11ec-8d3d-0242ac130003"]);
+      XCTAssertEqual(discount.signature, @"test_signature");
+      addPaymentTimes++;
+      return YES;
+    }
+    addPaymentTimes++;
+    return YES;
+  };
 
   FlutterError *error;
 
   [self.plugin addPaymentPaymentMap:argument error:&error];
+  XCTAssertEqual(addPaymentTimes, 1);
   XCTAssertNil(error);
-  OCMVerify(
-      times(1),
-      [mockHandler
-          addPayment:[OCMArg checkWithBlock:^BOOL(id obj) {
-            SKPayment *payment = obj;
-            if (@available(iOS 12.2, *)) {
-              SKPaymentDiscount *discount = payment.paymentDiscount;
-
-              return [discount.identifier isEqual:@"test_identifier"] &&
-                     [discount.keyIdentifier isEqual:@"test_key_identifier"] &&
-                     [discount.nonce
-                         isEqual:[[NSUUID alloc]
-                                     initWithUUIDString:@"4a11a9cc-3bc3-11ec-8d3d-0242ac130003"]] &&
-                     [discount.signature isEqual:@"test_signature"] &&
-                     [discount.timestamp isEqual:@(1635847102)];
-            }
-
-            return YES;
-          }]]);
 }
 
 - (void)testAddPaymentFailureWithInvalidPaymentDiscount {
   // Support for payment discount is only available on iOS 12.2 and higher.
   if (@available(iOS 12.2, *)) {
     NSDictionary *argument = @{
       @"productIdentifier" : @"123",
-      @"quantity" : @(1),
+      @"quantity" : @"123",  // This should normally be an int, not a string
```

Comment:
I am confused - the value here is a string. 

also why do we need behavioral change in this kind of refactor work?

Reply:
yea, the test is testing invalid payment types. it should be an int, but i gave it a string, so its now an invalid payment.
The original tests stubs out the translator so it returns an error, but its nicer to directly make the payment invalid and test the behaviour that way.

Reply:
can you add your explanation int he comment? From the existing comment it's pretty hard to understand what you intend to do

Reply:
okay sureee